sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/BlobAsyncClientBase.java

```diff
@@ -230,6 +230,9 @@ protected BlobAsyncClientBase(HttpPipeline pipeline, String url, BlobServiceVers
             throw LOGGER.logExceptionAsError(
                 new IllegalArgumentException("'snapshot' and 'versionId' cannot be used at the same time."));
         }
+        if (blobName == null) {
+            throw LOGGER.logExceptionAsError(new IllegalArgumentException("'blobName' can not be set to null."));
```

Comment:
For null values, [SDK guidelines](https://azure.github.io/azure-sdk/java_introduction.html#java-errors-system-errors) suggest that we throw NullPointerException.